homeassistant/components/epson/media_player.py

```diff
@@ -82,22 +92,30 @@
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Epson media player platform."""
-    if DATA_EPSON not in hass.data:
-        hass.data[DATA_EPSON] = []
-
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
-    ssl = config[CONF_SSL]
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Epson projector from a config entry."""
 
+    timeout_scale = config_entry.options.get(TIMEOUT_SCALE, 1.0)
     epson_proj = EpsonProjector(
-        async_get_clientsession(hass, verify_ssl=False), name, host, port, ssl
+        async_get_clientsession(
+            hass, verify_ssl=config_entry.data.get(CONF_SSL, False)
+        ),
+        config_entry.title,
+        config_entry.data[CONF_HOST],
+        config_entry.data[CONF_PORT],
+        timeout_scale,
+        config_entry.entry_id,
     )
+    if not hass.data.get(DOMAIN):
+        hass.data[DOMAIN] = {config_entry.entry_id: {}}
+    if DATA_EPSON not in hass.data:
+        hass.data[DATA_EPSON] = []
 
+    hass.data[DOMAIN][config_entry.entry_id] = config_entry.add_update_listener(
+        update_listener
+    )
```

Comment:
Also move to `__init__.py`. You also need to unregister from the update listener upon unload.